packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -115,6 +121,58 @@ class DatePickerInput extends React.Component {
     }
   }
 
+  static getDerivedStateFromProps(props, state) {
+    // Formats the selected date into a consistent value and sets input state.
+    if (props.value !== state.prevDate) {
+      const format = DateUtil.getDateFormat(props.intl.formatMessage({ id: 'Terra.datePicker.dateFormatOrder' }));
+      const tempDate = DateUtil.convertToISO8601(props.value, format);
+      return {
+        day: tempDate.substring(8, 10),
+        month: tempDate.substring(5, 7),
+        year: tempDate.substring(0, 4),
+        prevDate: props.value,
+      };
+    }
+    return null;
+  }
+
+  /**
+   * Moves focus to the correct input depending on date ordering. Focus changing is
+   * disabled if a complete date has been entered in order to make single input
+   * corrections easier, and is re-enabled if the whole date is erased.
+   * @param {string} inputValue - The value from the current input.
+   * @param {int} type - The input type, based on DateUtil.inputType.
+   * @param {string} dateVariant - Date variant based on the currently supported date
+   *                      orders: 'MM-DD-YYYY', 'DD-MM-YYYY', or 'YYYY-MM-DD'.
+   */
+  moveFocusOnChange(inputValue, type, dateVariant) {
```

Comment:
Should the internals of this method be broken out into the DateUtils so this can be unit tested?